modules/xfem/src/geomsearch/XFEMElementPairLocator.C

```diff
@@ -41,5 +46,41 @@ XFEMElementPairLocator::reinit()
 
     ElementPairInfo new_elem_info(elem, q_points, weights, normal);
     _element_pair_info.insert(std::pair<const Elem*, ElementPairInfo>(elem, new_elem_info));
+
+    // elem2
+    _xfem->getXFEMIntersectionInfo(elem2, plane_id, normal, intersectionPoints, _use_displaced_mesh);
+
+    if (intersectionPoints.size() == 2)
+      _xfem->getXFEMqRuleOnLine(intersectionPoints, q_points, weights);
+    else if (intersectionPoints.size() > 2)
+      _xfem->getXFEMqRuleOnSurface(intersectionPoints, q_points, weights);
+
+    ElementPairInfo new_elem_info2(elem2, q_points, weights, normal);
+    _element_pair_info.insert(std::pair<const Elem*, ElementPairInfo>(elem2, new_elem_info2));
```

Comment:
I see you point. Do you think if I should add a class/struct inside ```ElementPairInfo``` to hold the quadrature info for each element? Or if you have any better ideas?